src/test/kotlin/com/charleskorn/kaml/YamlReadingTest.kt

```diff
@@ -857,7 +977,55 @@ object YamlReadingTest : Spek({
                 context("parsing that input") {
                     it("throws an appropriate exception") {
                         assert({ Yaml.default.parse(ComplexStructure.serializer(), input) }).toThrow<MalformedYamlException> {
-                            message { toBe("Property name must not be a list, map or null value. (To use 'null' as a property name, enclose it in quotes.)") }
+                            message { toBe("Property name must not be a list, map, null or tagged value. (To use 'null' as a property name, enclose it in quotes.)") }
+                            line { toBe(1) }
+                            column { toBe(1) }
+                        }
+                    }
+                }
+            }
+
+            context("given some input representing an object with a null as a key") {
+                val input = """
+                    null: something
+                """.trimIndent()
+
+                context("parsing that input") {
+                    it("throws an appropriate exception") {
+                        assert({ Yaml.default.parse(ComplexStructure.serializer(), input) }).toThrow<MalformedYamlException> {
+                            message { toBe("Property name must not be a list, map, null or tagged value. (To use 'null' as a property name, enclose it in quotes.)") }
+                            line { toBe(1) }
+                            column { toBe(1) }
+                        }
+                    }
+                }
+            }
+
+            context("given some input representing an object with an object as a key") {
+                val input = """
+                    { }: something
+                """.trimIndent()
+
+                context("parsing that input") {
+                    it("throws an appropriate exception") {
+                        assert({ Yaml.default.parse(ComplexStructure.serializer(), input) }).toThrow<MalformedYamlException> {
+                            message { toBe("Property name must not be a list, map, null or tagged value. (To use 'null' as a property name, enclose it in quotes.)") }
+                            line { toBe(1) }
+                            column { toBe(1) }
+                        }
+                    }
+                }
+            }
+
+            context("given some input representing an object with a tagged value as a key") {
+                val input = """
+                    !<sealedInt> { }: something
```

Comment:
To make the intent of this test clear, could you remove the `{ }` and replace it with a string (eg. `!<sealedInt> thing: something`)?

Reply:
That's not yet possible, but I can create another PR after that one. Currently scalar nodes with tags are not supported and supporting them would be a new feature with a new PR in my opinion.